frontend/app/i18n-config.server.ts

```diff
@@ -14,13 +14,15 @@ import { getLanguage } from '~/utils/i18n-utils';
  *
  * @param languageOrRequest - The language code or Request object to get the language from.
  * @param namespace - The namespace to get the translation function for.
+ * @param keyPrefix - The key prefix to use for the translation function.
  * @returns A translation function for the given language and namespace.
  * @throws {AppError} If no language is found in the `languageOrRequest`.
  */
-export async function getFixedT<NS extends Namespace>(
+export async function getFixedT<NS extends Namespace, TKPrefix extends KeyPrefix<NS> = undefined>(
   languageOrRequest: Language | Request,
   namespace: NS,
-): Promise<TFunction<NS>> {
+  keyPrefix?: TKPrefix,
```

Comment:
I originally didn't add this to the function because I feared that it would lead to inconsistent code.. in some places we'd be using a prefix: `t('namespace:this.is.my.property')` and in other places we'd be omitting the prefix: `t('namespace:property')` ← what's the full key name? 🤷

It doesn't hurt anything to have it here, but I would highly discourage its use for code readability reasons.